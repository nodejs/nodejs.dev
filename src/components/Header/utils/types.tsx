export enum HeaderType { Link, Anchor }

export interface LinkHeaderProps {
  type: HeaderType,
  to: string,
  classes: string,
  title: string,
  activeClassName: string,
  partiallyActive: boolean
}

export interface AnchorHeaderProps {
  type: HeaderType,
  href: string,
  classes: string,
  title?: string,
  mobileTitle?: string,
  checkMobile?: boolean
}
