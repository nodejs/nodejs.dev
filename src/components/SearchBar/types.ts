import type { SerialisedIndexData } from 'elasticlunr';
import type { SearchResult } from '../../types';

export interface SearchProps {
  localSearchLearnPages: {
    index: SerialisedIndexData<SearchResult>;
  };
}
