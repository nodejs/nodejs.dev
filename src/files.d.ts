declare module '*.svg' {
  export const ReactComponent: React.FC<React.SVGProps<SVGSVGElement>>;
  const svgUrl: string;
  export default svgUrl;
}

declare module '*.png';
declare module '*.jpg';
