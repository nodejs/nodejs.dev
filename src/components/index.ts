import * as ArticleComponents from './ArticleComponents';
import * as BlogComponents from './BlogComponents';
import * as CommonComponents from './CommonComponents';
import * as DownloadComponents from './DownloadComponents';
import * as NavigationComponents from './NavigationComponents';

import { getApiComponents, Components as ApiComponents } from './ApiComponents';

export {
  ApiComponents,
  ArticleComponents,
  BlogComponents,
  CommonComponents,
  DownloadComponents,
  NavigationComponents,
  getApiComponents,
};
