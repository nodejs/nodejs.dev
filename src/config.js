/**
 * Common App Configurations.
 */

module.exports = {
  title: 'Node.js',
  description:
    "Node.jsÂ® is a JavaScript runtime built on Chrome's V8 JavaScript engine.",
  siteUrl: 'https://nodejs.dev',
  siteUrlNoSlash: 'https://nodejs.dev',
  featuredImage: 'https://nodejs.org/static/images/logo-hexagon-card.png',
  color: '#6cc24a',
  icon: 'src/images/favicon.png', // This path is relative to the root of the site.
  lang: 'en',
  display: 'browser',
  ogType: 'website',
  ogImgType: 'image/png',
  ogImgWidth: '224',
  ogImgHeight: '256',
  twitter: '@nodejs',
  twitterCard: 'summary',
  twitterImgAlt: 'The Node.js Hexagon Logo',
  facebook: 'https://www.facebook.com/nodejsfoundation/',
};
