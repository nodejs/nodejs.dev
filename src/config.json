{
  "title": "Node.js",
  "description": "Node.jsÂ® is a JavaScript runtime built on Chrome's V8 JavaScript engine.",
  "siteUrl": "https://nodejs.dev",
  "siteUrlNoSlash": "https://nodejs.dev",
  "featuredImage": "https://nodejs.dev/static/images/og-image.png",
  "color": "#6cc24a",
  "icon": "src/images/favicon.png",
  "lang": "en",
  "display": "browser",
  "ogType": "website",
  "ogImgType": "image/png",
  "ogImgWidth": "224",
  "ogImgHeight": "256",
  "twitter": "@nodejs",
  "twitterCard": "summary",
  "twitterImgAlt": "The Node.js Hexagon Logo",
  "facebook": "https://www.facebook.com/nodejsfoundation/",
  "bannerBtnText": "Read more",
  "nodeGcalId": "nodejs.org_nr77ama8p7d7f9ajrpnu506c98@group.calendar.google.com"
}
