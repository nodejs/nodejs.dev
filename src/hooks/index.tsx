export { useApiData } from './useApiDocs';
export { useReleaseHistory } from './useReleaseHistory';
