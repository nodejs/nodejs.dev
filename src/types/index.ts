export interface HomepageData {
  page: {
    id: string;
    html: string;
    frontmatter: {
      title: string;
      displayTitle: string;
      subTitle: string;
      description: string;
      learnLinkText: string;
      beginnerGuideHeaderText: string;
      beginnerGuideBodyText: string;
      doMoreWithNodeHeaderText: string;
      doMoreWithNodeBodyText: string;
      nodeFeatureHeader1: string;
      nodeFeatureHeader2: string;
      nodeFeatureHeader3: string;
      nodeFeature1: string;
      nodeFeature2: string;
      nodeFeature3: string;
      nodeFeatureAltText: string;
    };
  };
}

export interface LearnPageContext {
  slug: string;
  relativePath: string;
  next: PaginationInfo;
  previous: PaginationInfo;
  navigationData: NavigationSectionData;
}

export interface LearnPageData {
  doc: {
    id: string;
    html: string;
    tableOfContents: string;
    frontmatter: { title: string; description: string };
    fields: { authors: string[] };
  };
}

export interface PaginationInfo {
  slug: string;
  title: string;
}

export interface NavigationSectionItem {
  slug: string;
  title: string;
  section: string;
}

export interface NavigationSectionData {
  [index: string]: NavigationSectionItem[];
}

/**
 * Payload on the `stickychange` custom event
 */
export interface StickyChange {
  stuck: boolean;
  target: HTMLElement;
}

export interface SentinelObserverSetupOptions {
  container: HTMLElement;
  stickyElementsClassName: string;
  root?: HTMLElement | null;
  headerRootMargin?: string;
  footerRootMargin?: string;
}

declare global {
  interface Window {
    previousPath: string;
  }
}

export interface Page {
  editPath?: string;
  data: {
    page: {
      html: string;
      tableOfContents: string;
      frontmatter: {
        title: string;
        description: string;
      };
      fields: { authors: string[] };
    };
  };
  location?: Location;
}
export interface CommunityNavigationSection {
  title: string;
  sections: string[];
}

export interface BlogPostAuthor {
  id?: string;
  name: string;
  url: string;
}

export interface BlogMetaData {
  node: {
    frontmatter: { title: string; author: BlogPostAuthor[] };
    fields: { date: string; slug: string };
  };
}
export interface BlogPostsList {
  blogs: {
    edges: BlogMetaData[];
  };
}

export interface BlogPageData {
  blog: {
    html: string;
    excerpt: string;
    frontmatter: { title: string; author: BlogPostAuthor[] };
    fields: { slug: string; date: string };
  };
  recent: {
    edges: BlogMetaData[];
  };
}

export interface BlogPageContext {
  slug: string;
  relativePath: string;
  next: PaginationInfo;
  previous: PaginationInfo;
  navigationData: NavigationSectionData;
}
