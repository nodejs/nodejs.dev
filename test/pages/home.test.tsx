import React from 'react';
import { render } from '@testing-library/react';
import Index from '../../src/pages';
import { HomepageData } from '../../src/types';

const homePageData: HomepageData = {
  page: {
    id: 'id-mock',
    html: 'html-mock',
    frontmatter: {
      title: 'title-mock',
      displayTitle: 'displayTitle-mock',
      subTitle: 'subTitle-mock',
      description: 'description-mock',
      learnLinkText: 'learnLinkText-mock',
      beginnerGuideHeaderText: 'beginnerGuideHeaderText-mock',
      beginnerGuideBodyText: 'beginnerGuideBodyText-mock',
      doMoreWithNodeHeaderText: 'doMoreWithNodeHeaderText-mock',
      doMoreWithNodeBodyText: 'doMoreWithNodeBodyText-mock',
      nodeFeatureHeader1: 'nodeFeatureHeader1-mock',
      nodeFeatureHeader2: 'nodeFeatureHeader2-mock',
      nodeFeatureHeader3: 'nodeFeatureHeader3-mock',
      nodeFeature1: 'nodeFeature1-mock',
      nodeFeature2: 'nodeFeature2-mock',
      nodeFeature3: 'nodeFeature3-mock',
      nodeFeatureAltText: 'nodeFeatureAltText-mock',
    },
  },
};

describe('Home page', () => {
  it('renders correctly', () => {
    const { container } = render(<Index data={homePageData} />);
    expect(container).toMatchSnapshot();
  });
});
