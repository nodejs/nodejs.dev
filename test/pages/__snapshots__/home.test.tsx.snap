// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders correctly 1`] = `
<div>
  <div
    class="layout-container"
  >
    <nav
      aria-label="Primary"
      class="nav"
    >
      <div
        class="nav__container"
      >
        <div
          class="nav__startwrapper"
        >
          <a
            href="/"
          >
            <div
              class="logo"
            >
              <img
                alt="Node.js"
                class="nav__logo light-mode-only"
                src="test-file-stub"
              />
              <img
                alt="Node.js"
                class="nav__logo dark-mode-only"
                src="test-file-stub"
              />
            </div>
          </a>
          <ul
            class="nav__tabs__container"
          >
            <li
              class="nav__tabs"
            >
              <a
                class="activeStyleTab"
                href="/learn"
              >
                Learn
              </a>
            </li>
            <li
              class="nav__tabs"
            >
              <a
                class="activeStyleTab"
                href="https://nodejs.org/en/docs/"
                rel="noopener noreferrer"
                target="_blank"
              >
                Documentation
              </a>
            </li>
            <li
              class="nav__tabs"
            >
              <a
                class="activeStyleTab"
                href="/download"
              >
                Download
              </a>
            </li>
            <li
              class="nav__tabs"
            >
              <a
                class="activeStyleTab"
                href="https://nodejs.org/en/get-involved/"
                rel="noopener noreferrer"
                target="_blank"
              >
                Community
              </a>
            </li>
          </ul>
        </div>
        <div
          class="nav__endwrapper"
        >
          <ul
            class="right-container"
          >
            <li
              class="nav__tabs nav__tabs--right"
            >
              <button
                class="dark-mode-toggle"
                type="button"
              >
                <span
                  class="sr-only"
                >
                  Toggle Dark Mode
                </span>
                <i
                  class="material-icons light-mode-only"
                >
                  nights_stay
                </i>
                <i
                  class="material-icons dark-mode-only"
                >
                  wb_sunny
                </i>
              </button>
            </li>
            <li
              class="nav__tabs"
            >
              <a
                href="https://github.com/nodejs/nodejs.dev"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  class="sr-only"
                >
                  GitHub
                </span>
                <svg
                  height="2.2rem"
                  style="fill: var(--color-text-accent);"
                  viewBox="0 0 438.549 438.549"
                  width="2.2rem"
                >
                  <path
                    d="M409.132,114.573c-19.608-33.596-46.205-60.194-79.798-79.8C295.736,15.166,259.057,5.365,219.271,5.365   c-39.781,0-76.472,9.804-110.063,29.408c-33.596,19.605-60.192,46.204-79.8,79.8C9.803,148.168,0,184.854,0,224.63   c0,47.78,13.94,90.745,41.827,128.906c27.884,38.164,63.906,64.572,108.063,79.227c5.14,0.954,8.945,0.283,11.419-1.996   c2.475-2.282,3.711-5.14,3.711-8.562c0-0.571-0.049-5.708-0.144-15.417c-0.098-9.709-0.144-18.179-0.144-25.406l-6.567,1.136   c-4.187,0.767-9.469,1.092-15.846,1c-6.374-0.089-12.991-0.757-19.842-1.999c-6.854-1.231-13.229-4.086-19.13-8.559   c-5.898-4.473-10.085-10.328-12.56-17.556l-2.855-6.57c-1.903-4.374-4.899-9.233-8.992-14.559   c-4.093-5.331-8.232-8.945-12.419-10.848l-1.999-1.431c-1.332-0.951-2.568-2.098-3.711-3.429c-1.142-1.331-1.997-2.663-2.568-3.997   c-0.572-1.335-0.098-2.43,1.427-3.289c1.525-0.859,4.281-1.276,8.28-1.276l5.708,0.853c3.807,0.763,8.516,3.042,14.133,6.851   c5.614,3.806,10.229,8.754,13.846,14.842c4.38,7.806,9.657,13.754,15.846,17.847c6.184,4.093,12.419,6.136,18.699,6.136   c6.28,0,11.704-0.476,16.274-1.423c4.565-0.952,8.848-2.383,12.847-4.285c1.713-12.758,6.377-22.559,13.988-29.41   c-10.848-1.14-20.601-2.857-29.264-5.14c-8.658-2.286-17.605-5.996-26.835-11.14c-9.235-5.137-16.896-11.516-22.985-19.126   c-6.09-7.614-11.088-17.61-14.987-29.979c-3.901-12.374-5.852-26.648-5.852-42.826c0-23.035,7.52-42.637,22.557-58.817   c-7.044-17.318-6.379-36.732,1.997-58.24c5.52-1.715,13.706-0.428,24.554,3.853c10.85,4.283,18.794,7.952,23.84,10.994   c5.046,3.041,9.089,5.618,12.135,7.708c17.705-4.947,35.976-7.421,54.818-7.421s37.117,2.474,54.823,7.421l10.849-6.849   c7.419-4.57,16.18-8.758,26.262-12.565c10.088-3.805,17.802-4.853,23.134-3.138c8.562,21.509,9.325,40.922,2.279,58.24   c15.036,16.18,22.559,35.787,22.559,58.817c0,16.178-1.958,30.497-5.853,42.966c-3.9,12.471-8.941,22.457-15.125,29.979   c-6.191,7.521-13.901,13.85-23.131,18.986c-9.232,5.14-18.182,8.85-26.84,11.136c-8.662,2.286-18.415,4.004-29.263,5.146   c9.894,8.562,14.842,22.077,14.842,40.539v60.237c0,3.422,1.19,6.279,3.572,8.562c2.379,2.279,6.136,2.95,11.276,1.995   c44.163-14.653,80.185-41.062,108.068-79.226c27.88-38.161,41.825-81.126,41.825-128.906   C438.536,184.851,428.728,148.168,409.132,114.573z"
                  />
                </svg>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <main
      class="home-page"
    >
      <div
        class="home-page-hero"
      >
        <h1>
          displayTitle-mock
        </h1>
        <h2
          class="sub-title t-subheading"
        >
          subTitle-mock
        </h2>
        <div
          class="btn-ctas"
        >
          <div
            class="download-lts-container"
          >
            <a
              class="circular-container"
              href="https://nodejs.org/dist//node-.tar.gz"
            >
              Download Node (LTS)
            </a>
            <p
              class="t-caption"
            >
              
              <a
                href="https://nodejs.org/dist//node-.tar.gz"
              >
                Get Current
              </a>
            </p>
          </div>
          <a
            class="circular-container inverse"
            href="/learn"
          >
            Learn Node
          </a>
        </div>
      </div>
      <section
        class="node-demo-container"
      >
        <div
          class="node-demo"
        >
          <div
            class="install"
          >
            <div
              class="react-tabs"
              data-tabs="true"
            >
              <div
                class="install__header"
              >
                <div
                  class="install__header-circles"
                >
                  <div
                    class="install__header-grey-circle"
                  />
                  <div
                    class="install__header-grey-circle"
                  />
                  <div
                    class="install__header-grey-circle"
                  />
                </div>
                <div
                  class="install__header-text"
                >
                  bash
                </div>
              </div>
              <ul
                class="react-tabs__tab-list"
                role="tablist"
              >
                <li
                  aria-controls="react-tabs-1"
                  aria-disabled="false"
                  aria-selected="true"
                  class="react-tabs__tab react-tabs__tab--selected"
                  id="react-tabs-0"
                  role="tab"
                  tabindex="0"
                >
                  Windows (Chocolatey)
                </li>
                <li
                  aria-controls="react-tabs-3"
                  aria-disabled="false"
                  aria-selected="false"
                  class="react-tabs__tab"
                  id="react-tabs-2"
                  role="tab"
                >
                  macOS (nvm)
                </li>
                <li
                  aria-controls="react-tabs-5"
                  aria-disabled="false"
                  aria-selected="false"
                  class="react-tabs__tab"
                  id="react-tabs-4"
                  role="tab"
                >
                  Linux (nvm)
                </li>
              </ul>
              <div
                aria-labelledby="react-tabs-0"
                class="react-tabs__tab-panel react-tabs__tab-panel--selected"
                id="react-tabs-1"
                role="tabpanel"
              >
                <div>
                  <pre
                    class="shell-box"
                  >
                    <div
                      class="shell-box-top"
                    >
                      <span>
                        SHELL
                      </span>
                      <button
                        type="button"
                      >
                        copy
                      </button>
                    </div>
                    <code>
                      <span
                        class="install__text__no-select"
                      >
                        $
                      </span>
                      <span
                        class="install__text__command"
                      >
                        choco 
                      </span>
                      install nodejs-lts
                    </code>
                  </pre>
                  <br />
                  <br />
                  <a
                    class="install__docs-button"
                    href="https://nodejs.org/en/download/package-manager/#windows"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Read documentation
                  </a>
                </div>
              </div>
              <div
                aria-labelledby="react-tabs-2"
                class="react-tabs__tab-panel"
                id="react-tabs-3"
                role="tabpanel"
              />
              <div
                aria-labelledby="react-tabs-4"
                class="react-tabs__tab-panel"
                id="react-tabs-5"
                role="tabpanel"
              />
            </div>
          </div>
        </div>
        <img
          alt=""
          class="leafs-front animations"
          src="test-file-stub"
        />
        <img
          alt=""
          class="leafs-middle animations"
          src="test-file-stub"
        />
        <img
          alt=""
          class="leafs-back animations"
          src="test-file-stub"
        />
        <img
          alt=""
          class="dots"
          src="test-file-stub"
        />
      </section>
      <section
        class="node-features"
      >
        <div
          class="node-features__feature"
        >
          <svg
            alt="Node Feature"
            class="feature-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M255.917 480a32.536 32.536 0 0 1-16.633-4.599l-52.985-32.44c-7.914-4.562-4.023-6.203-1.443-7.141 10.565-3.781 13.713-5.657 24.947-12.285 1.206-.667 2.747-.424 3.955.322l39.71 23.504c1.476.85 3.557.85 4.931 0l155.188-92.246c1.475-.877 2.415-2.646 2.415-4.441V163.869c0-1.85-.94-3.592-2.449-4.528l-155.12-94.672c-1.478-.894-3.421-.894-4.898 0L98.516 159.374c-1.544.903-2.516 2.698-2.516 4.495v186.805c0 1.813.972 3.513 2.481 4.389l39.929 23.972c23.61 12.204 37.59-.17 37.59-14.611V180.725c0-2.652 2.047-4.727 4.596-4.727h22.809c2.515 0 4.597 2.072 4.597 4.727v183.698c0 32.563-19.353 51.248-49.199 51.248-9.156 0-16.397 0-36.552-10.279l-41.584-24.781C70.371 374.459 64 362.965 64 350.656V161.191c0-12.316 6.371-23.784 16.665-29.917L239.35 36.41c10.027-5.88 23.374-5.88 33.332 0l158.65 94.864C441.63 137.423 448 148.899 448 161.191v189.465c0 12.309-6.37 23.75-16.668 29.953l-158.65 94.774a32.52 32.52 0 0 1-16.698 4.599l-.067.018z"
            />
            <path
              d="M304.943 351.998c-64.61 0-84.006-31.61-84.006-59.271 0-2.629 2.048-4.729 4.562-4.729h20.521c2.282 0 4.227 1.7 4.562 4.016 3.084 21.602 16.748 31.15 54.324 31.15 33.399 0 47.091-10.346 47.091-28.684 0-10.592-3.463-18.424-55.407-23.697-43.427-4.441-70.288-14.373-70.288-50.295 0-33.135 26.996-52.49 72.234-52.49 46.128 0 76.462 14 79.173 50.829.102 1.337-.368 2.629-1.241 3.644-.871.965-2.078 1.527-3.353 1.527h-20.591c-2.146 0-4.024-1.562-4.459-3.713-4.401-16.953-16.97-23.402-49.563-23.402-36.486 0-40.746 12.753-40.746 22.607 0 11.963 5.031 15.441 54.294 22.172 48.761 6.663 71.933 16.117 71.933 51.552 0 35.781-28.808 58.783-79.075 58.783l.035.001z"
            />
          </svg>
          <h4>
            JavaScript
          </h4>
          <p>
            Node.js provides support for the JavaScript programming language
          </p>
        </div>
        <div
          class="node-features__feature"
        >
          <svg
            alt="Node Feature"
            class="feature-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 376.6V206.8c0-46.4-3.7-70.8-22.1-94C374 87.7 348.9 77 310 77h-22V16l-96 96 96 96v-65h22c24.6 0 29 3.6 33.8 9.6 5.5 6.9 8.2 19.1 8.2 54.2v169.8c-19.1 11.1-32 31.7-32 55.4 0 35.3 28.7 64 64 64s64-28.7 64-64c0-23.7-12.9-44.3-32-55.4zM384 472c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zM128 48c-35.3 0-64 28.7-64 64 0 23.7 12.9 44.3 32 55.4v209.2c-19.1 11.1-32 31.7-32 55.4 0 35.3 28.7 64 64 64s64-28.7 64-64c0-23.7-12.9-44.3-32-55.4V167.4c19.1-11.1 32-31.7 32-55.4 0-35.3-28.7-64-64-64zm0 424c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zm0-320c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40z"
            />
          </svg>
          <h4>
            Open Source
          </h4>
          <p>
            Node.js is open source and actively maintained by contributors all over the world
          </p>
        </div>
        <div
          class="node-features__feature"
        >
          <svg
            alt="Node Feature"
            class="feature-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 421.6c-18.1 0-33.2-6.8-42.9-10.9-5.4-2.3-11.3 1.8-10.9 7.6l3.5 51c.2 3.1 3.8 4.7 6.3 2.8l14.5-11c1.8-1.4 4.5-.9 5.7 1l20.5 32.1c1.5 2.4 5.1 2.4 6.6 0l20.5-32.1c1.2-1.9 3.9-2.4 5.7-1l14.5 11c2.5 1.9 6.1.3 6.3-2.8l3.5-51c.4-5.8-5.5-10-10.9-7.6-9.8 4.1-24.8 10.9-42.9 10.9z"
            />
            <path
              d="M397.7 293.1l-48-49.1c0-158-93.2-228-93.2-228s-94.1 70-94.1 228l-48 49.1c-1.8 1.8-2.6 4.5-2.2 7.1L130.6 412c.9 5.7 7.1 8.5 11.8 5.4l67.1-45.4s20.7 20 47.1 20c26.4 0 46.1-20 46.1-20l67.1 45.4c4.6 3.1 10.8.3 11.8-5.4l18.5-111.9c.2-2.6-.6-5.2-2.4-7zM256.5 192c-17 0-30.7-14.3-30.7-32s13.8-32 30.7-32c17 0 30.7 14.3 30.7 32s-13.7 32-30.7 32z"
            />
          </svg>
          <h4>
            Everywhere
          </h4>
          <p>
            Node.js has been adapted to work in a wide variety of places
          </p>
        </div>
      </section>
    </main>
    <footer
      class="footer"
    >
      <ul
        class="footer__left"
      >
        <li>
          <a
            class="footer__link"
            href="https://nodejs.org/en/about/trademark/"
          >
            Trademark Policy
          </a>
        </li>
        <li>
          <a
            class="footer__link"
            href="https://nodejs.org/en/about/privacy/"
          >
            Privacy Policy
          </a>
        </li>
        <li>
          <a
            class="footer__link"
            href="https://github.com/openjs-foundation/cross-project-council/blob/master/CODE_OF_CONDUCT.md#contributor-covenant-code-of-conduct"
          >
            Code of Conduct
          </a>
        </li>
        <li>
          <a
            class="footer__link"
            href="https://nodejs.org/en/security/"
          >
            Security Reporting
          </a>
        </li>
        <li>
          <a
            class="footer__link"
            href="https://nodejs.org/en/about/"
          >
            About
          </a>
        </li>
        <li>
          <a
            class="footer__link"
            href="https://nodejs.org/en/blog/"
          >
            Blog
          </a>
        </li>
      </ul>
      <ul
        class="footer__right"
      >
        <li>
          © OpenJS Foundation
        </li>
        <li>
          <a
            aria-label="Node.js Github Page Link"
            href="https://github.com/nodejs/node"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              class="sr-only"
            >
              GitHub
            </span>
            <svg
              height="1.8rem"
              style="fill: var(--color-text-secondary);"
              viewBox="0 0 438.549 438.549"
              width="1.8rem"
            >
              <path
                d="M409.132,114.573c-19.608-33.596-46.205-60.194-79.798-79.8C295.736,15.166,259.057,5.365,219.271,5.365   c-39.781,0-76.472,9.804-110.063,29.408c-33.596,19.605-60.192,46.204-79.8,79.8C9.803,148.168,0,184.854,0,224.63   c0,47.78,13.94,90.745,41.827,128.906c27.884,38.164,63.906,64.572,108.063,79.227c5.14,0.954,8.945,0.283,11.419-1.996   c2.475-2.282,3.711-5.14,3.711-8.562c0-0.571-0.049-5.708-0.144-15.417c-0.098-9.709-0.144-18.179-0.144-25.406l-6.567,1.136   c-4.187,0.767-9.469,1.092-15.846,1c-6.374-0.089-12.991-0.757-19.842-1.999c-6.854-1.231-13.229-4.086-19.13-8.559   c-5.898-4.473-10.085-10.328-12.56-17.556l-2.855-6.57c-1.903-4.374-4.899-9.233-8.992-14.559   c-4.093-5.331-8.232-8.945-12.419-10.848l-1.999-1.431c-1.332-0.951-2.568-2.098-3.711-3.429c-1.142-1.331-1.997-2.663-2.568-3.997   c-0.572-1.335-0.098-2.43,1.427-3.289c1.525-0.859,4.281-1.276,8.28-1.276l5.708,0.853c3.807,0.763,8.516,3.042,14.133,6.851   c5.614,3.806,10.229,8.754,13.846,14.842c4.38,7.806,9.657,13.754,15.846,17.847c6.184,4.093,12.419,6.136,18.699,6.136   c6.28,0,11.704-0.476,16.274-1.423c4.565-0.952,8.848-2.383,12.847-4.285c1.713-12.758,6.377-22.559,13.988-29.41   c-10.848-1.14-20.601-2.857-29.264-5.14c-8.658-2.286-17.605-5.996-26.835-11.14c-9.235-5.137-16.896-11.516-22.985-19.126   c-6.09-7.614-11.088-17.61-14.987-29.979c-3.901-12.374-5.852-26.648-5.852-42.826c0-23.035,7.52-42.637,22.557-58.817   c-7.044-17.318-6.379-36.732,1.997-58.24c5.52-1.715,13.706-0.428,24.554,3.853c10.85,4.283,18.794,7.952,23.84,10.994   c5.046,3.041,9.089,5.618,12.135,7.708c17.705-4.947,35.976-7.421,54.818-7.421s37.117,2.474,54.823,7.421l10.849-6.849   c7.419-4.57,16.18-8.758,26.262-12.565c10.088-3.805,17.802-4.853,23.134-3.138c8.562,21.509,9.325,40.922,2.279,58.24   c15.036,16.18,22.559,35.787,22.559,58.817c0,16.178-1.958,30.497-5.853,42.966c-3.9,12.471-8.941,22.457-15.125,29.979   c-6.191,7.521-13.901,13.85-23.131,18.986c-9.232,5.14-18.182,8.85-26.84,11.136c-8.662,2.286-18.415,4.004-29.263,5.146   c9.894,8.562,14.842,22.077,14.842,40.539v60.237c0,3.422,1.19,6.279,3.572,8.562c2.379,2.279,6.136,2.95,11.276,1.995   c44.163-14.653,80.185-41.062,108.068-79.226c27.88-38.161,41.825-81.126,41.825-128.906   C438.536,184.851,428.728,148.168,409.132,114.573z"
              />
            </svg>
          </a>
        </li>
        <li>
          <a
            aria-label="Node.js Twitter Link"
            href="https://twitter.com/nodejs"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              height="1.8rem"
              style="fill: var(--color-text-secondary);"
              viewBox="328 355 335 276"
              width="1.8rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d=" M 630, 425 A 195, 195 0 0 1 331, 600 A 142, 142 0 0 0 428, 570 A  70,  70 0 0 1 370, 523 A  70,  70 0 0 0 401, 521 A  70,  70 0 0 1 344, 455 A  70,  70 0 0 0 372, 460 A  70,  70 0 0 1 354, 370 A 195, 195 0 0 0 495, 442 A  67,  67 0 0 1 611, 380 A 117, 117 0 0 0 654, 363 A  65,  65 0 0 1 623, 401 A 117, 117 0 0 0 662, 390 A  65,  65 0 0 1 630, 425 Z"
              />
            </svg>
          </a>
        </li>
        <li>
          <a
            aria-label="Node.js Slack Link"
            href="https://slack.openjsf.org"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              height="1.8rem"
              style="fill: var(--color-text-secondary);"
              viewBox="0 0 54 54"
              width="1.8rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19.712.133a5.381 5.381 0 0 0-5.376 5.387 5.381 5.381 0 0 0 5.376 5.386h5.376V5.52A5.381 5.381 0 0 0 19.712.133m0 14.365H5.376A5.381 5.381 0 0 0 0 19.884a5.381 5.381 0 0 0 5.376 5.387h14.336a5.381 5.381 0 0 0 5.376-5.387 5.381 5.381 0 0 0-5.376-5.386"
              />
              <path
                d="M53.76 19.884a5.381 5.381 0 0 0-5.376-5.386 5.381 5.381 0 0 0-5.376 5.386v5.387h5.376a5.381 5.381 0 0 0 5.376-5.387m-14.336 0V5.52A5.381 5.381 0 0 0 34.048.133a5.381 5.381 0 0 0-5.376 5.387v14.364a5.381 5.381 0 0 0 5.376 5.387 5.381 5.381 0 0 0 5.376-5.387"
              />
              <path
                d="M34.048 54a5.381 5.381 0 0 0 5.376-5.387 5.381 5.381 0 0 0-5.376-5.386h-5.376v5.386A5.381 5.381 0 0 0 34.048 54m0-14.365h14.336a5.381 5.381 0 0 0 5.376-5.386 5.381 5.381 0 0 0-5.376-5.387H34.048a5.381 5.381 0 0 0-5.376 5.387 5.381 5.381 0 0 0 5.376 5.386"
              />
              <path
                d="M0 34.249a5.381 5.381 0 0 0 5.376 5.386 5.381 5.381 0 0 0 5.376-5.386v-5.387H5.376A5.381 5.381 0 0 0 0 34.25m14.336-.001v14.364A5.381 5.381 0 0 0 19.712 54a5.381 5.381 0 0 0 5.376-5.387V34.25a5.381 5.381 0 0 0-5.376-5.387 5.381 5.381 0 0 0-5.376 5.387"
              />
            </svg>
          </a>
        </li>
      </ul>
    </footer>
  </div>
</div>
`;
